include/h2o/memory.h

```diff
@@ -63,6 +63,17 @@ extern "C" {
 #define H2O_NORETURN
 #endif
 
+#if defined(__clang__) && H2O_GNUC_VERSION >= 0x20500
+#define H2O_NORETURN_PTR __attribute__((noreturn))
+#elif __STDC_VERSION__ >= 201112L
+#define H2O_NORETURN_PTR _Noreturn
+#elif defined(__GNUC__) && H2O_GNUC_VERSION >= 0x20500
+// noreturn was not defined before gcc 2.5
+#define H2O_NORETURN_PTR __attribute__((noreturn))
+#else
+#define H2O_NORETURN_PTR
+#endif
```

Comment:
Am I correct to assume that `__attribute__((noreturn))` is a superset of `_Noreturn` for both GCC and Clang, with the difference that the former can be used for function pointers too?

If so, would it be possible to change the definition of `H2O_NORETURN` (the original one) to prefer `__attribute__((noreturn))` over `_Noreturn` for the two compilers, and call it a day?

Reply:
it might be the case, here's the docs I found below. Simply using `__attribute__((noreturn))` may be the way to go.

related docs:
https://clang.llvm.org/docs/AttributeReference.html#noreturn-noreturn
https://gcc.gnu.org/onlinedocs/gcc/Common-Function-Attributes.html
https://en.cppreference.com/w/c/language/attributes/noreturn

> The [[_Noreturn]] spelling is deprecated and only exists to ease code migration for code using [[noreturn]] after including <stdnoreturn.h>.

Reply:
Thank you for checking!